code/Test_definitions/Location-Retrieval.feature

```diff
@@ -0,0 +1,264 @@
+  Scenario: The device value is an empty object
+    Given the request body property "$.device" is set to empty object
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  @location_retrieval_11_device_schema_compliant
+  # Test every type of identifier even if not supported by the implementation
+  Scenario Outline: Some device identifier value does not comply with the schema
+    Given the request body property "<device_identifier>" does not comply with the OAS schema at "<oas_spec_schema>"
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+    Examples:
+      | device_identifier          | oas_spec_schema                             |
+      | $.device.phoneNumber       | /components/schemas/PhoneNumber             |
+      | $.device.ipv4Address       | /components/schemas/NetworkAccessIdentifier |
+      | $.device.ipv6Address       | /components/schemas/DeviceIpv4Addr          |
+      | $.device.networkIdentifier | /components/schemas/DeviceIpv6Address       |
+
+  @location_retrieval_11.1_device_phoneNumber_schema_compliant
+  # Example of the scenario above with a higher level of specification
+  # TBD if test plan has to provide specific testing values to provoke an error
+  Scenario Outline: Device identifier phoneNumber value does not comply with the schema
+    Given the request body property "$.device.phoneNumber" is set to: <phone_number_value>
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+    Examples:
+      | phone_number_value |
+      | string_value       |
+      | 1234567890         |
+      | +12334foo22222     |
+      | +00012230304913849 |
+      | 123                |
+      | ++49565456787      |
+
+  @location_retrieval_12_device_identifiers_unsupported
+  Scenario: None of the provided device identifiers is supported by the implementation
+    Given that some type of device identifiers are not supported by the implementation
+    And the request body property "$.device" only includes device identifiers not supported by the implementation
+    When the HTTP "POST" request is sent
+    Then the response status code is 422
+    And the response property "$.status" is 422
+    And the response property "$.code" is "UNSUPPORTED_DEVICE_IDENTIFIERS"
+    And the response property "$.message" contains a user friendly text
+
+  @location_retrieval_13_device_not_found

```

Comment:
Will this 404 case be possible in three-legged token use cases? The access token would be tied to a valid device already?

Reply:
There may be situations as @bigludo7 suggests when the access token is valid but the device included in the request does not point to any device. 

* Device can NOT be deducted from the access token:
  * `device` NOT included in the request body-> 422 UNIDENTIFIABLE_DEVICE
  * `device` included in the request body but not pointing to a known device -> 404 DEVICE_NOT_FOUND